lib/internal/process.js

```diff
@@ -22,6 +23,53 @@ const assert = process.assert = function(x, msg) {
 };
 
 
+// Set up the process.cpuUsage() function.
+function setup_cpuUsage() {
+  // Get the native function, which will be replaced with a JS version.
+  const _cpuUsage = process.cpuUsage;
+
+  // Create the argument array that will be passed to the native function.
+  const cpuValues = new Float64Array(2);
+
+  // Replace the native function with the JS version that calls the native
+  // function.
+  process.cpuUsage = function cpuUsage(prevValue) {
+    // Call the native function to get the current values
+    if (_cpuUsage(cpuValues) !== 0) {
+      throw new Error('unable to obtain CPU usage time');
+    }
+
+    // If no previous value passed in, return current value.
+    if (prevValue == null) return {
+      user: cpuValues[0],
+      system: cpuValues[1]
+    };
+
+    // If a previous value was passed in, ensure it has the correct shape.
+    if (!previousValueIsValid(prevValue.user)) {
+      throw new TypeError('value of user property of argument is invalid');
+    }
+
+    if (!previousValueIsValid(prevValue.system)) {
+      throw new TypeError('value of system property of argument is invalid');
```

Comment:
checking user input should be the first thing that happens. If it's invalid then there's no point in calling into c++.